OSSM-8849 New Features (#89195)
openshift/openshift-docs: modules/ossm-release-notes-3-0-new-features.adoc

== Support for Istioctl for select platforms and commands

This release adds support for select platforms and commands for Istioctl, the command line utility for the Istio project that includes many diagnostic and debugging utilities. For more information, see "Support for Istioctl".

[IMPORTANT]
====
Installing Istio using the `istioctl` utility is not supported.
====

[id="support-for-multi-cluster_{context}"]
== Support for multi-cluster

This release introduces support for the Istio multi-cluster deployment models:

* Multi-primary
* Primary-remote
* An external control plane

The federation feature introduced in {SMProduct} 2.1 is not available in {SMPRoduct} 3.0.

[id="multiple-control-planes-in-a-single-cluster_{context}"]
== Multiple control planes in a single cluster

This release adds support for the Istio feature of multiple control planes in a single cluster. This replaces the `MultiTenant` deployment model (mode) in {SMProduct} 2.

[id="revision-based-upgrades_{context}"]
== Revision based upgrades

{SMProduct} 3 retains support for simple in-place style updates, and adds support for canary-style updates of the Istio control plane using Istio's revision feature.

> The `Istio` resource manages Istio revision labels using the `IstioRevision` resource. When the Istio resource's `updateStrategy` type is set to `RevisionBased`, it creates Istio revision labels using the Istio resource's name combined with the Istio version, for example `mymesh-v1-21-2`.

Reviewer: `manages Istio revision labels using the` I don't think Istio is managing/creating revision labels. I'm not sure what was the idea here, probably to describe how the `Istiorevision` name is generated?

////
Module included in the following assemblies:
* service-mesh-docs-main/ossm-release-notes-assembly.adoc
////

:_mod-docs-content-type: REFERENCE
[id="ossm-release-3-0-new-features-enhancements_{context}"]
= {SMProduct} 3.0 new features and enhancements

> This release makes {SMProductName} 3 generally available, adds new features, addresses Common Vulnerabilities and Exposures (CVEs), and is supported on {ocp-product-title} 4.14 and later.

Reviewer 2: Sometimes, throughout this document, we say "3" and other times we say "3.0". Suggest consistency.

> This release makes {SMProductName} 3 generally available, adds new features, addresses Common Vulnerabilities and Exposures (CVEs), and is supported on {ocp-product-title} 4.14 and later.

Author: Service Mesh does not link to errata.

> For a list of supported component versions and support features, see "Service Mesh 3.0 feature support tables".

Author: It's in the "Additional resources" section.

For a complete list of changes between {SMProduct} 2 and {SMProduct} 3, see "Important information to know if you are migrating from OpenShift Service Mesh 2.6".

[id="migration-guides_{context}"]
== Migration guides

This release adds a set of checklists and migration guides to help you migrate from {SMProduct} 2 to {SMProduct} 3.

You must complete the checklists first. The checklists help you set up and configure {SMProduct} 2 and the `ServiceMeshControlPlane` resource to migrate to {SMProduct} 3 and the `Istio` control plane resource.

Your migration depends on your deployment model:

* Multitenant
* Multitenant with cert-manager
* Cluster-wide
* Cluster-wide with cert-manager

You can also migrate gateways.

[id="new-istio-distribution-and-operator_{context}"]
== New Istio distribution and operator

> {SMProduct} 3.0 is based on a Red{nbsp}Hat distribution of the link:https://istio.io/[Istio project] and is deployed with a new operator for Istio based on the Sail Operator project that is part of the **istio-ecosystem organization** on GitHub. The Sail Operator includes a new set of custom resource definitions (CRDs) for managing Istio. For example, the `Istio` CRD replaces the `ServiceMeshControlPlane` CRD in previous releases of {SMProduct}.

Author: Keeping it **istio-ecosystem** to be consistent with how it is referenced throughout 3.x docs.

[id="support-for-istoctl-for-select-platforms-and-commands_{context}"]
== Support for Istioctl for select platforms and commands

This release adds support for select platforms and commands for Istioctl, the command line utility for the Istio project that includes many diagnostic and debugging utilities. For more information, see "Support for Istioctl".

[IMPORTANT]
====
Installing Istio using the `istioctl` utility is not supported.
====

[id="support-for-multi-cluster-deployment-models_{context}"]
== Support for multi-cluster deployment models

This release introduces support for the following Istio multi-cluster deployment models:

* Multi-primary
* Primary-remote
* An external control plane

> The federation feature introduced in {SMProduct} 2.1 is not available in {SMProduct} 3.0.

Author: It is not deprecate or removed. It is developer preview but not shipped with 3.0 so it is not available for 3.0.

[id="multiple-control-planes-in-a-single-cluster_{context}"]
== Multiple control planes in a single cluster

This release adds support for the Istio feature of multiple control planes in a single cluster. This replaces the `MultiTenant` deployment model (mode) in {SMProduct} 2.

[id="revision-based-updates_{context}"]
== Revision based updates

This release adds support for canary-style updates of the Istio control plane using the Istio revision feature. This enables a new Istio control plane to be created alongside the existing Istio control plane so that workloads can be migrated incrementally. The update strategy is configured using the `spec.updateStrategy` parameter of the `Istio` resource.

> For more information, see "About RevisionBased strategy".

Reviewer 2: Missing link/xref.

Author: Added to "Additional resources." Thanks!

[id="istiocni-custom-resource-definition_{context}"]
== IstioCNI custom resource definition

> This release introduces the `IstioCNI` custom resource definition (CRD), which is used to manage the lifecycle of the Istio Container Network Interface (CNI) daemon set. A single instance of this resource must be created per cluster to configure traffic redirection for pods in the mesh. The Istio CNI lifecycle is independent of the Istio control plane(s).

Reviewer 2: https://www.ibm.com/docs/en/ibm-style?topic=grammar-plurals#-s-

```suggestion
This release introduces the `IstioCNI` custom resource definition (CRD), which is used to manage the lifecycle of the Istio Container Network Interface (CNI) daemon set. A single instance of this resource must be created per cluster to configure traffic redirection for pods in the mesh. The Istio CNI lifecycle is independent of the Istio control plane or planes.
```

[id="dual-stack-ipv4-ipv6_{context}"]
== IPv4/IPv6 dual stack (Technology Preview)

> :FeatureName: IPv4/IPv6 dual stack

Reviewer 2: I typically see this as "dual-stack" but I can't find any rule on this...

```suggestion
:FeatureName: IPv4/IPv6 dual-stack
```

Author: Istio refers to it as dual-stack, and Dual Stack ... but it looks like Dual Stack is for headings, and dual-stack in text. Will update references to dual-stack. Thanks!

[id="dual-stack-ipv4-ipv6_{context}"]
== IPv4/IPv6 dual stack (Technology Preview)
:FeatureName: IPv4/IPv6 dual stack
include::snippets/technology-preview.adoc[]

This release includes IPv4/IPv6 dual stack support as a technology preview feature. This aligns with the 'Alpha' status of the Istio upstream project, and is feature-complete for Istio when using sidecars. Dual stack helps organizations smoothly transition to IPv6, while still maintaining compatibility with their existing IPv4 setup.

In {SMProduct} 3.0, dual stack is disabled by default in the `Istio` resource. You can enable it with specific configuration changes, such as the one shown in the following example:

.Example YAML configuration for IPv4/IPv6 Dual Stack
[source,yaml]
----
apiVersion: sailoperator.io/v1
kind: Istio
metadata:
  name: default
spec:
  values:
    meshConfig:
      defaultConfig:
        proxyMetadata:
          ISTIO_DUAL_STACK: "true"
    pilot:
      ipFamilyPolicy: RequireDualStack
      env:
        ISTIO_DUAL_STACK: "true"

>   version: v1.24.3

Reviewer 3: Let's remove the version from the yaml. 

```suggestion
```

[id="dual-stack-ipv4-ipv6_{context}"]
== IPv4/IPv6 dual stack (Technology Preview)
:FeatureName: IPv4/IPv6 dual stack
include::snippets/technology-preview.adoc[]

This release includes IPv4/IPv6 dual stack support as a technology preview feature. This aligns with the 'Alpha' status of the Istio upstream project, and is feature-complete for Istio when using sidecars. Dual stack helps organizations smoothly transition to IPv6, while still maintaining compatibility with their existing IPv4 setup.

In {SMProduct} 3.0, dual stack is disabled by default. You can enable it with specific configuration changes, such as the ones shown in the following example:

.Example configuration for IPv4/IPv6 Dual Stack
[source,yaml]
----
spec:
  values:
    meshConfig:
      defaultConfig:
        proxyMetadata:
          ISTIO_DUAL_STACK: "true"
    pilot:
      ipFamilyPolicy: RequireDualStack
      env:
        ISTIO_DUAL_STACK: "true"
----

[id="istio-ambient-mode_{context}"]
== Istio Ambient mode (Developer Preview)
:FeatureName: Istio Ambient mode
include::snippets/developer-preview.adoc[]

> Istio Ambient Mode provides a sidecar-less service mesh architecture that reduces resource overhead, simplifies operations, and allows incremental adoption without application changes. It maintains security and observability through a layered security model with mTLS and authorization. OpenShift Service Mesh 3 operator includes deploying the Ambient profile as a developer preview feature using the community Ztunnel image. However, the ambient profile should not be used on clusters with production workloads or for multi-control plane use cases. Additionally, the community Ztunnel image does not work on IBM Power and Z Platforms.

Reviewer 4: We should add to the list that the ZTunnel image will also not work on "OpenShift Container Platform clusters in FIPS mode"

Source: https://redhat-internal.slack.com/archives/C01R9E3SFEV/p1740762618836279?thread_ts=1740759024.029839&cid=C01R9E3SFEV
